deployments/azure/templates/init/network.bicep

```diff
@@ -1,29 +1,47 @@
 param networkName string
 param networkCidr array
 param subnetCidr string
+param albSubnetCidr string
+param deployAlbSubnet bool
 param location string
 param tags object = {}
 
+var gwSubnetName = '${networkName}-gtw-subnet'
+
 resource network 'Microsoft.Network/virtualNetworks@2022-11-01' = {
   name: networkName
   location: location
-  tags: tags
+
   properties: {
     addressSpace: {
       addressPrefixes: networkCidr
     }
   }
+  tags: tags
 }
 
 resource subnet 'Microsoft.Network/virtualNetworks/subnets@2022-11-01' = {
   parent: network
   name: networkName
+
   properties: {
     addressPrefix: subnetCidr
     privateEndpointNetworkPolicies: 'Enabled'
     privateLinkServiceNetworkPolicies: 'Enabled'
   }
 }
 
-output subnetName string = subnet.name
+resource gwSubnet 'Microsoft.Network/virtualNetworks/subnets@2022-11-01' = if (deployAlbSubnet) { 
```

Comment:
I'm using the resources arm template to deploy with, and I enabled deployAlbSubnet, and keep running in this issue:
![Screenshot 2024-04-15 at 10 47 43 AM](https://github.com/Teradata/ai-unlimited/assets/166751279/046c82ed-611c-474f-951c-1fe2e07bab27)

It is consistent for me, and I have run with different `name` values. Are you able to reproduce this issue?